synapse_invite_checker/invite_checker.py

```diff
@@ -845,11 +903,32 @@ async def room_scan(self) -> None:
 
         logger.debug("Detected %d total rooms", len(all_room_ids))
 
-        rooms_to_purge = set()
+        server_notice_rooms = set()
+        # On servers that don't have server notices set up, the mxid will be None
+        if self.api._hs.config.servernotices.server_notices_mxid is not None:
+            # This database call was chosen over `is_server_notice_room()` as batching
+            # the data should be much more efficient than calling the latter for every
+            # single room
+            server_notice_rooms_list: list[RoomsForUser] = (
+                await self.api._store.get_rooms_for_local_user_where_membership_is(
+                    self.api._hs.config.servernotices.server_notices_mxid,
+                    [Membership.JOIN],
+                    [],
+                )
+            )
+            # The server_notice_rooms_list is more complex than we need, pare it down to
+            # room_id's only. A set() is conveniently O(1) complexity. Also probably not
+            # a good idea to just remove these outright from the complete room list, to
+            # still check all the inactive rooms to purge below
+            server_notice_rooms = set(room.room_id for room in server_notice_rooms_list)
```

Comment:
```suggestion
            server_notice_rooms = {room.room_id for room in server_notice_rooms_list}
```

Probably makes no difference, but afaik we use set literals elsewhere.

Reply:
Yeah, sounds good